pycbc/waveform/waveform.py

```diff
@@ -801,6 +801,18 @@ def imrphenomd_length_in_time(**kwds):
     """
     return get_imr_length("IMRPhenomD", **kwds)
 
+def imrphenomhm_length_in_time(**kwargs):
+    """Estimates the duration of IMRPhenom waveforms that include higher modes.
+    """
+    if 'mode_array' in kwargs and kwargs['mode_array'] is not None:
+        maxm = max(m for _, m in kwargs['mode_array'])
+    else:
+        # the highest m for all of these is 4 (from the 4,4 mode)
+        maxm = 4
+    # we'll use the PhenomD length, then scale it by maxm / 2
+    dur = get_imr_length("IMRPhenomD", **kwargs)
+    return dur * maxm/2.
```

Comment:
I don't think this is the right equation. You should modidfy get_imr_length to give you the duration from f_lower / (maxm/2)). The duration doesn't just scale.

Reply:
Good point. I had tested a few points using the m/2 thing, and it seemed to work. You're right that scaling the frequency makes more sense. See the update.